File: Easy/1732_Find_the_Highest_Altitude/Solution.py
from typing import List, Optional


class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        maxNum = 0
        prefixSum = 0

        for i in range(len(gain)):
            prefixSum += gain[i]
            maxNum = max(maxNum, prefixSum)

        return maxNum


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    gain1 = [-5,1,5,0,-7]
    print(solution.largestAltitude(gain1))

    # test cases 2
    gain2 = [-4,-3,-2,-1,4,3,2]
    print(solution.largestAltitude(gain2))
    

File: Easy/1732_Find_the_Highest_Altitude/Solution.java



public class Solution {
    public int largestAltitude(int[] gain) {
        int maxNum = 0;
        int prefixSum = 0;

        for (int i = 0; i < gain.length; i++) {
            prefixSum += gain[i];
            maxNum = Math.max(maxNum, prefixSum);
        }

        return maxNum;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        int[] gain1 = {-5,1,5,0,-7};
        System.out.println(solution.largestAltitude(gain1));

        // test cases 2
        int[] gain2 = {-4,-3,-2,-1,4,3,2};
        System.out.println(solution.largestAltitude(gain2));
        
    }
}